### ISSUE TITLE
'Welcome Guide' menu item uses a wrong component thus triggering confusing `speak` messages

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#52328]
### Description

In the Post Editor and in the Widgets page, when opening the welcome guides from the 'Options' > 'Welcome guide' mehu item, an audible `speak` message is triggered. The text of the message is:

`Preference activated - Welcome Guide`

which is pretty confusing, as opening the welcome guide isn't really activating a preference.

Turns out in the Post Editor and in the Widgets page the 'Welcome guide' menu item uses the `PreferenceToggleMenuItem` component under the hoods, which is incorrect. It should use a normal menuitem.
-  The`PreferenceToggleMenuItem` component is meant to be used for _preferences_ that can be activated and deactivated, typically: Top toolbar, Spotlight mode, Fullscreen mode, Distraction free.
- he`PreferenceToggleMenuItem` [automatically triggers a speak message](https://github.com/WordPress/gutenberg/blob/e244388d8669618b76c966cc33d48df9156c2db6/packages/preferences/src/components/preference-toggle-menu-item/index.js#L31-L51) with the label of the menu item, which is supposed to be a preference and certainly not a welcome guide.
-  The`PreferenceToggleMenuItem` component ultimately renders a `MenuItem` with `role="menuitemcheckbox"`, which is a menu item that can be checked / unchecked (sort of a checkbox), which is not appropriate for a control that just opens a modal dialog.

The`PreferenceToggleMenuItem` component should only be used for preferences that can be activated, persisted, and deactivated. Toggling modal dialogs should use a simple menuitem, ideally with an [aria-haspopup="dialog"](https://www.w3.org/TR/wai-aria-1.2/#aria-haspopup) attribute.

### Step-by-step reproduction instructions

- Use a screen reader or observe the content of the element with ID `a11y-speak-polite` in your browser dev tools inspector
- In the Post editor, Click: Options > Welcome guide
- Observe a speak message is triggered: `Preference activated - Welcome Guide`
- In the Settings sidebar > Post tab, click the template name then click 'Edit template`.
- The editor switches to 'template editing' mode.
- Click: Options > Welcome guide
- Observe a speak message is triggered: `Preference activated - Welcome Guide`
- Activate a non-block based theme e.g. Twenty Twenty-One
- Go to Appearance > Widgets
- Click: Options > Welcome guide
- Observe a speak message is triggered: `Preference activated - Welcome Guide`

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Considering also https://github.com/WordPress/gutenberg/issues/51702 looks like there's the need of new compoennts for this menu item:
- They should use the editor store and check for the current editor mode.
- Based on the current editor mode, they should open the right Welcome guide.
- They should be simple `role="menuitem"` instead of `role="menuitemcheckbox"`.
- The rendered menu item should use an [aria-haspopup="dialog"](https://www.w3.org/TR/wai-aria-1.2/#aria-haspopup) attribute.
- They should not trigger any `speak` message.
